src/Sentry/Scope.cs

```diff
@@ -197,7 +197,24 @@ public void AddBreadcrumb(Breadcrumb breadcrumb)
         /// <summary>
         /// Adds an attachment.
         /// </summary>
-        public void AddAttachment(Attachment attachment) => _attachments.Add(attachment);
+        public void AddAttachment(Attachment attachment)
+        {
+            // Limit file size
+            if (attachment.Length > Options.MaxAttachmentSize)
+            {
+                Options.DiagnosticLogger?.LogWarning(
+                    "Attachment '{0}' dropped because it's too large.",
+                    attachment.FileName
```

Comment:
How large?

```suggestion
                    "Attachment '{0}' dropped because it's too large {1}.",
                    attachment.FileName, attachment.Length
```